src/Config/SimpleClientConfig.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Cake\Queue\Config;
+
+class SimpleClientConfig
+{
+    private $simpleClientConfig = [];
+
+    private $queue = 'default';
+
+    /**
+     * __construct
+     *
+     * @param  mixed $queue the name of the queue drawn from Queue.default.queue or worker -Q queuename
+     * @param  mixed $config the Queue.{configname} array where {configname} is the config name i.e. default
+     * @return void
+     */
+    public function __construct(string $queue, array $config)
+    {
+        $this->create($queue, $config);
+    }
+
+    /**
+     * get
+     * return the SimpleClientConfig array
+     *
+     * @return array
+     */
+    public function get(): array
+    {
+        return $this->simpleClientConfig;
+    }
+
+    /**
+     * getQueue
+     * Return the queue name
+     *
+     * @return string
+     */
+    public function getQueue(): string
+    {
+        return $this->queue;
+    }
+
+    /**
+     * create
+     * Creates the config array to pass to SimpleClient(...)
+     *
+     * @param  mixed $queue queue name
+     * @param  mixed $config the Queue.configname array
+     * @return void
+     */
+    public function create($queue, $config)
+    {
+        $this->queue = $queue !== 'default' ? $queue : $config['queue'];
+
+        $this->simpleClientConfig = [
+            'transport' => $config['url'],
```

Comment:
I don't think this will work for all the adapters, as the URL string can contain more information than just the transport name.

Reply:
I will have a further look at trying to pass args via the url config key.

Reply:
Can we add a flag to the Config array and use the dsn or the config array if needed?

```
'Queue' => [
    'default' => [
        // A DSN for your configured backend. default: null
        'url' => 'beanstalk:',
        // pass as a config array instead of transport dsn
        'useConfigArray' => true,
        'configArray' => [
            'transport' => 'beanstalk:',
            'client' =>
            [
                'prefix' => 'enqueue',
                'separator' => '.',
                'app_name' => 'app',
                'router_topic' => 'default',
                'router_queue' => 'default',
                'default_queue' => 'default',
            ],
            'extensions' => [
                'signal_extension' => false,
                'reply_extension' => false,
            ],
        ],
```